Automattic/jetpack
# Blocks: switch to new shared, extracted implementation of Tracks

#### Changes proposed in this Pull Request:

**Do not merge, this is waiting on #27937**

This is a follow-up to #27937. Let's start using our shared implementation.

#### Other information:


#### Jetpack product discussion

This will benefit all blocks and more plugins (like Jetpack Social), but I'm building it as part of 1386-gh-dotcom-forge

#### Does this pull request change what data or activity we track or use?

It doesn't change data we track; it will continue to be tracked all the same.

#### Testing instructions:

* Start by enabling upgrade nudges on your test site, it will make it easier to test things: add `add_filter( 'jetpack_block_editor_enable_upgrade_nudge', '__return_true' );` to a functionality plugin.
* Open up the block editor in your browser, open the js console, and send the following: `localStorage.setItem( 'debug', 'dops:analytics*' );`. Refresh the page for good measure.
* In the js console settings, check "Preserve log"
* From that point on, you should see events being logged when you take some actions with Jetpack blocks, like:
    - Add a Paid block like Payments on your free Jetpack site.
    - Publish a post, and click on the Anchor button in the post-publish panel.
    - And other events as seen in this PR.
